distribution/lib/Standard/Base/0.0.0-dev/src/System/File.enso

```diff
@@ -685,19 +694,20 @@ type File
              example_list_md_files =
                  Examples.data_dir.list name_filter="**.{txt,md}" recursive=True
     list : Text -> Boolean -> Vector File
-    list self name_filter=Nothing recursive=False =
+    list self name_filter:Text="" recursive:Boolean=False =
         all_files = case recursive of
             True -> list_descendants self
             False -> self.list_immediate_children
-        filtered = case name_filter of
-            Nothing -> all_files
+        case name_filter of
+            "" -> all_files
             _ ->
+                used_filter = if recursive.not || name_filter.contains "**" then name_filter else
```

Comment:
contains or starts_with? If it has a `**` in the middle but not at the start, will it still work right?

Reply:
if it has `**` in the middle I am trusting you know what you are doing!